Source/swiftlint/Extensions/LinterCache+CommandLine.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  LinterCache+CommandLine.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 12/31/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SwiftLintFramework
+
+extension LinterCache {
+
+    class func makeCache(options: LintOptions, configuration: Configuration) -> LinterCache? {
+        guard let url = cacheURL(options: options, configuration: configuration) else {
+            return nil
+        }
+
+        let configurationHash = configuration.hash
+        let cache: LinterCache
+        do {
+            cache = try LinterCache(contentsOf: url, configurationHash: configurationHash)
+        } catch {
+            cache = LinterCache(configurationHash: configurationHash)
+        }
+
+        return cache
+    }
+
+    func save(options: LintOptions, configuration: Configuration) {
+        if let url = cacheURL(options: options, configuration: configuration) {
+            try? save(to: url)
+        }
+    }
+
+}
+
+private func cacheURL(options: LintOptions, configuration: Configuration) -> URL? {
+    guard !options.ignoreCache else {
+        return nil
+    }
+
+    let path = options.cachePath.isEmpty ? configuration.cachePath : options.cachePath
+    return path.map(URL.init(fileURLWithPath:)) ?? defaultCacheURL(options: options)
+}
+
+private func defaultCacheURL(options: LintOptions) -> URL {
```

Comment:
Much of this should be moved into SwiftLintFramework IMHO, so that the cache is used even if not going through the SwiftLint CLI.